### ISSUE TITLE
Add Health Checks

### ISSUE THREAD

**Issue by hishamco** [OrchardCMS/OrchardCore#13588]
Recently I created two demos as a part of Orchard Core Contrib (OCC), which added some health checks to OC, IMHO there're many checks we might add to Orchard Core modules OOB as the following:

- [ ] Tenants Health Check
- [x] Redis Health Check #13589
- [ ] Database Health Check
- [ ] Email Health Check
- [ ] SMS Health Check

Also, we could add

- [ ] Localization Health Check
- [ ] Content Health Check

and much more, feel free to add to this list if the checks are useful

Another thing we can add as a part of the health check

- [ ] Health Checks UI
- [x] Health Check Response #13590

**Comment by hishamco** [comment]
I will create small manageable PRs are usual to make it easier to track this enhancement

**Comment by sebastienros** [comment]
Redis, Db, Email for sure because these are external services. The internal one (localization, content, ...) I don't know what would be "checked" because the app would not even work, which means there might be an internal HC that would already fail. Like the HC endpoint itself.

**Comment by Piedone** [comment]
What do you mean by Tenants Health Check? Thought about having an aggregating feature that basically flips the Default tenant to Unhealthy if any tenant is Unhealthy.

**Comment by hishamco** [comment]
Nope, but it shows the health of the current tenant. I need to revise this

**Comment by Piedone** [comment]
OK, then that's done.
